source/operators/captions/webcaptions.py

```diff
@@ -0,0 +1,1016 @@
+            name = operator_name+"_"+self.source_language_code
+        except KeyError:
+            self.operator_object.update_workflow_status("Error")
+            self.operator_object.add_workflow_metadata(WebCaptionsError="Missing language code for WebCaptions {e}".format(e=e))
+            raise MasExecutionError(self.operator_object.return_output_object())
+
+        print("WebCaptionsOperatorName() Name {}".format(name))
+        return name
+
+    def CaptionsOperatorName(self, language_code=None):
+
+        # Shouldn't assume WebCaptions operator is WebCaptions, maybe pass it in the configuration?
+        operator_name = "Captions"
+
+        if language_code != None:
+            return operator_name+"_"+language_code
+
+        try:
+            name = operator_name+"_"+self.source_language_code
+        except KeyError:
+            self.operator_object.update_workflow_status("Error")
+            self.operator_object.add_workflow_metadata(WebCaptionsError="Missing language code for WebCaptions {e}".format(e=e))
+            raise MasExecutionError(self.operator_object.return_output_object())
+
+        print("CaptionsOperatorName() Name {}".format(name))
+        return name
+
+
+
+    def GetTranscript(self):
+
+        transcript = []
+
+        response = dataplane.retrieve_asset_metadata(self.asset_id, operator_name=self.transcribe_operator_name)
+        transcript.append(response["results"])
+
+        while "cursor" in response:
+            response = dataplane.retrieve_asset_metadata(self.asset_id, operator_name=self.transcribe_operator_name, cursor=response["cursor"])
+            transcript.append(response["results"])
+
+        return transcript
+
+    def TranscribeToWebCaptions(self, transcripts):
+
+        endTime = 0.0
+        maxLength = 120
+        wordCount = 0
+        maxWords = 25
+        maxSilence = 1.5
+
+        captions = []
+        caption = None
+
+        for transcript in transcripts:
+            for item in transcript["results"]["items"]:
+
+                isPunctuation = item["type"] == "punctuation"
+
+                if caption is None:
+
+                    # Start of a line with punctuation, just skip it
+                    if isPunctuation:
+                        continue
+
+                    # Create a new caption line
+                    caption = {
+                        "start": float(item["start_time"]),
+                        "caption": "",
+                        "wordConfidence": []
+                    }
+
+                if not isPunctuation:
+
+                    startTime = float(item["start_time"])
+
+                    # Check to see if there has been a long silence
+                    # between the last recorded word and start a new
+                    # caption if this is the case, ending the last time
+                    # as this one starts.
+
+                    if (len(caption["caption"]) > 0) and ((endTime + maxSilence) < startTime):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`.

 [Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)

Similar issue at line number 167.

Reply:
We are not going to address this CodeGuru finding because we want to mirror the official AWS code sample for sigv4 signing: https://docs.aws.amazon.com/code-samples/latest/catalog/python-signv4-v4-signing-get-post.py.html.